.github/workflows/main.yml

```diff
@@ -60,9 +66,11 @@ jobs:
     steps:
       - uses: actions/checkout@v2
       - name: Use Node ${{ matrix.node-version }}
-        uses: actions/setup-node@v1
+        uses: actions/setup-node@v4
         with:
           node-version: ${{ matrix.node-version }}
+          cache-dependency-path: extension/package-lock.json
```

Comment:
Could you help me understand why we need to cache dependency path here?

Reply:
Sure, no problem! The [setup-node action](https://github.com/actions/setup-node/tree/v4) can automatically help us cache Node.js dependencies. But to do so, it will try to find the lock file (package-lock.json) in the project's root by default. However, we don't have this file at the project's root, so we need to specify where the dependency file is located. Otherwise, the action will throw an error saying the dependencies lock file was not found (like [here](https://github.com/microsoft/vscode-gradle/actions/runs/6621380546/job/17985401242#step:4:25)).

We can remove the cache feature as well, but caching will help subsequent builds on CI to run faster 🙂

Reply:
Thank you for the explanation!